www/js/services.js

```diff
@@ -428,6 +429,162 @@ angular.module('emission.services', ['emission.plugin.logger'])
+      regionName: "European Union",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.14515,
+        BUS:         0.04751,
+        TRAIN:       0.048,
+        AIR_OR_HSR:  0.201
+      }
+    },
+    DE: {
+      regionName: "Germany",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.139,   // Tremod (passenger car)
+        BUS:         0.0535,  // Tremod (average city/coach)
+        TRAIN:       0.048,   // Tremod (average short/long distance)
+        AIR_OR_HSR:  0.201    // Tremod (airplane)
+      }
+    },
+    FR: {
+      regionName: "France",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.13125, // HBEFA (passenger car, considering 1 passenger)
+        BUS:         0.04838, // HBEFA (average short/long distance, considering 16/25 passengers)
+        TRAIN:       0.048,   // Tremod (DE value)
+        AIR_OR_HSR:  0.201    // Tremod (airplane)
+      }
+    },
+    AT: {
+      regionName: "Austria",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.14351, // HBEFA (passenger car, considering 1 passenger)
+        BUS:         0.04625, // HBEFA (average short/long distance, considering 16/25 passengers)
+        TRAIN:       0.048,   // Tremod (DE value)
+        AIR_OR_HSR:  0.201    // Tremod (airplane)
+      }
+    },
+    SE: {
+      regionName: "Sweden",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.13458, // HBEFA (passenger car, considering 1 passenger)
+        BUS:         0.04557, // HBEFA (average short/long distance, considering 16/25 passengers)
+        TRAIN:       0.048,   // Tremod (DE value)
+        AIR_OR_HSR:  0.201    // Tremod (airplane)
+      }
+    },
+    NO: {
+      regionName: "Norway",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.13265, // HBEFA (passenger car, considering 1 passenger)
+        BUS:         0.04185, // HBEFA (average short/long distance, considering 16/25 passengers)
+        TRAIN:       0.048,   // Tremod (DE value)
+        AIR_OR_HSR:  0.201    // Tremod (airplane)
+      }
+    },
+    CH: {
+      regionName: "Switzerland",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.17638, // HBEFA (passenger car, considering 1 passenger)
+        BUS:         0.04866, // HBEFA (average short/long distance, considering 16/25 passengers)
+        TRAIN:       0.048,   // Tremod (DE value)
+        AIR_OR_HSR:  0.201    // Tremod (airplane)
+      }
+    }
+  };
+
+  var defaultCarbonDatasetCode = 'US';
+  var currentCarbonDatasetCode;
+
+  this.initialize = function() {
+    currentCarbonDatasetCode = defaultCarbonDatasetCode;
+    var value = KVStore.getDirect(CARBON_DATASET_KEY);
```

Comment:
```suggestion
     return KVStore.get(CARBON_DATASET_KEY).then(function(value) {
          // stuff to do with the value. This will run asynchronously,
          // but it doesn't look like you rely on this for any reason
          // if you do, then you can just `then` on the returned promise to execute the dependencies **after** this is loaded.
         Logger.log("CarbonDatasetHelper.initialize() obtained value from storage [" + value + "]");
       if (!value) {
          value = defaultCarbonDatasetCode;
          Logger.log("CarbonDatasetHelper.initialize() no value in storage, using [" + value + "] instead");
            }
            this.setCurrentCarbonDatasetLocale(value); 
        });
```

Reply:
Does this work?

Reply:
Actually, wait. You do use this to return values from `getCurrentCarbonDatasetName`. Let me add the suggestions there too...